test/dotnet-new3.UnitTests/DotnetNew3CompleteTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using ApprovalTests;
+using FluentAssertions;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.TemplateEngine.TestHelper;
+using NuGet.Versioning;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Dotnet_new3.IntegrationTests
+{
+    public class DotnetNew3CompleteTests
+    {
+        private readonly ITestOutputHelper _log;
+
+        public DotnetNew3CompleteTests(ITestOutputHelper log)
+        {
+            _log = log;
+        }
+
+        [Fact]
+        public void CanDoTabCompletion()
+        {
+            string homeDir = TestUtils.CreateTemporaryFolder();
+            var commandResult = new DotnetNewCommand(_log, "complete", $"new3 --debug:custom-hive {homeDir} ")
+                .WithoutCustomHive()
+                .Execute();
+
+            commandResult
+                .Should()
+                .ExitWith(0)
+                .And.NotHaveStdErr();
+
+            Approvals.Verify(commandResult.StdOut);
+        }
+
+#pragma warning disable xUnit1004 // Test methods should not be skipped
+        [Fact(Skip = "not working")]
```

Comment:
https://github.com/dotnet/command-line-api/blob/d6046b4cbe51af1dfcf862fb64d844a8ab1bdb76/src/System.CommandLine/Parsing/ParseResultExtensions.cs#L274 invalidly determines symbol to complete here.
It determines it as `--language` option, which is already complete and at different position.

Likely this condition is wrong https://github.com/dotnet/command-line-api/blob/d6046b4cbe51af1dfcf862fb64d844a8ab1bdb76/src/System.CommandLine/Parsing/ParseResultExtensions.cs#L353-L356; it doesn't take into account if option has complete args set if position is specified.